### ISSUE TITLE
`redis-stack-server.service` removed but not replaced during update on Ubuntu 22.04

### ISSUE THREAD

**Issue by TimothyBramlett** [redis-stack/redis-stack#441]
### Issue

I recently updated the `redis-stack-server` package from version `6.2.6-v7` to version `6.2.6-v9` on Ubuntu 22.04. After the update, I noticed that the `redis-stack-server.service` file was removed from the `/etc/systemd/system` directory and was not replaced.

### Steps to Reproduce

1. Install `redis-stack-server` version `6.2.6-v7` on Ubuntu 22.04.
2. Update the package to version `6.2.6-v9`.
3. After the update, check the `/etc/systemd/system` directory for `redis-stack-server.service`.

### Expected Result

The `redis-stack-server.service` file is updated and remains in the `/etc/systemd/system` directory.

### Actual Result

The `redis-stack-server.service` file is removed from the `/etc/systemd/system` directory and is not replaced.

### Additional Information

I'm not sure why this is happening. As far as I know, the `redis-stack-server.service` file was originally installed automatically when I installed the `redis-stack-server` package.

I'd appreciate any insights or help in resolving this issue.  For now I will just copy it over manually and enable it.

**Comment by mika** [comment]
Yeah, same here also with the upgrade to 7.2.0-v0 on Debian/bullseye.

The problem is within the `postrm` maintainer script:

```
% cat /var/lib/dpkg/info/redis-stack-server.postrm 
#!/bin/sh

if [ -f /etc/systemd/system/redis-stack-server.service ]; then
    rm -f /etc/systemd/system/redis-stack-server.service
fi

if [ -f /etc/systemd/system/redis-stack.service ]; then
    rm -f /etc/systemd/system/redis-stack.service
fi

if [ -f /etc/systemd/system/redis-stack-redis.service ]; then
    rm -f /etc/systemd/system/redis-stack-redis.service
fi

if [ -f /etc/systemd/system/redisinsight.service ]; then
    rm -f /etc/systemd/system/redisinsight.service
fi
```

This postrm maintenance file is missing proper upgrade/removal handling, the common solution to get rid of configuration files on package removals (actually: purges!) is to check for it via:

```
case $1 in
  purge)
    $do_whatever_is_needed_on_package_removals
    ;;
esac
```

And if files need to be removed only for certain versions, then go through `dpkg-maintscript-helper rm_conffile ...`, see  https://www.debian.org/doc/debian-policy/ch-maintainerscripts.html and https://wiki.debian.org/DpkgConffileHandling for details.

**Comment by mika** [comment]
> Yeah, same here also with the upgrade to 7.2.0-v0 on Debian/bullseye.
> 
> The problem is within the `postrm` maintainer script:
> 
[...]

This bug is still present with 7.2.0-v4, though the fix should be quite easy.
Would you be willing to accept an PR if I'd provide one, @chayim?

**Comment by kissifrot** [comment]
Bumping this since it's still in 7.2.0-v9 :sweat_smile:

**Comment by martindale** [comment]
Experiencing this on Ubuntu 22.04

**Comment by TimothyBramlett** [comment]
@martindale Yeah who knows maybe it will be fixed in 2025

**Comment by wayne530** [comment]
Still happening in 7.2.0_v10-1. Additionally, `/etc/redis-stack.conf` is silently overwritten on update instead of the standard rpmnew/rpmsave-type behavior.

**Comment by blauweb** [comment]
I am another victim of this error.

**Comment by cokesaeba** [comment]
same on debian 11 last update... (2025, jan).

**Comment by eValker** [comment]
Same oh rhel8/9 while updating
`redis-stack-server-7.4.0_v4-1.rhel8.x86_64` 
to
`redis-stack-server-7.4.0_v5-1.rhel8.x86_64`

**Comment by thinkst-devdua** [comment]
Same with `redis-stack-server/jammy 7.4.0-v8 amd64`
